streams-tests/shared/src/test/scala/zio/stream/experimental/ZStreamSpec.scala

```diff
@@ -0,0 +1,76 @@
+package zio.stream.experimental
+
+import ZStreamUtils.nPulls
+
+import zio._
+import zio.test.Assertion.{ equalTo, isFalse, isTrue }
+import zio.test._
+
+object ZStreamSpec extends ZIOBaseSpec {
+  def spec = suite("ZStreamSpec")(
+    suite("Combinators")(
+      testM("map") {
+        ZStream
+          .fromEffect(UIO.succeed(1))
+          .map(_.toString)
+          .process
+          .use(nPulls(_, 3))
+          .map(assert(_)(equalTo(List(Right("1"), Left(Right(())), Left(Right(()))))))
+      }
+    ),
+    suite("Constructors")(
+      suite("fromEffect")(
+        testM("success") {
+          ZStream
+            .fromEffect(UIO.succeed(1))
+            .process
+            .use(nPulls(_, 3))
+            .map(assert(_)(equalTo(List(Right(1), Left(Right(())), Left(Right(()))))))
+        },
+        testM("failure") {
+          ZStream
+            .fromEffect(IO.fail("Ouch"))
+            .process
+            .use(nPulls(_, 3))
+            .map(assert(_)(equalTo(List(Left(Left("Ouch")), Left(Right(())), Left(Right(()))))))
+        }
+      ),
+      suite("managed")(
+        testM("success") {
```

Comment:
I like these tests, we should try to thoroughly exercise behavior (especially edge cases) now that we have clear semantics for everything.

Reply:
I had these tests before, check `StreamPullSafetySpec.scala`. :)